services/src/main/java/org/keycloak/cookie/CookieSecureResolver.java

```diff
@@ -0,0 +1,39 @@
+package org.keycloak.cookie;
+
+import jakarta.ws.rs.core.NewCookie;
+import org.keycloak.models.KeycloakContext;
+import org.keycloak.models.RealmModel;
+
+import java.net.URI;
+
+class CookieSecureResolver {
+
+    private final KeycloakContext context;
+    private final boolean sameSiteLegacyEnabled;
+
+    CookieSecureResolver(KeycloakContext context, boolean sameSiteLegacyEnabled) {
+        this.context = context;
+        this.sameSiteLegacyEnabled = sameSiteLegacyEnabled;
+    }
+
+    boolean resolveSecure(NewCookie.SameSite sameSite) {
+        // Due to cookies with SameSite=none secure context is always required when same-site legacy cookies are disabled
+        if (!sameSiteLegacyEnabled) {
+            return true;
+        } else {
```

Comment:
No need to have an `else` clause here, as the `if` is already terminated by a `return`.

Reply:
Sure, but personally I find this more readable, so keeping it as is